### PR TITLE
Add optional Cypher validation tool

### PR DESCRIPTION
**Pull request by tomasonjo**
LLMs have trouble with consistently getting the relationship direction accurately. That's why I organized a competition how to best and most simple to fix it based on the existing schema as a post-processing step. https://github.com/tomasonjo/cypher-direction-competition

I am adding the winner's code in this PR: https://github.com/sakusaku-rich/cypher-direction-competition

### CODE REVIEW

File: libs/langchain/langchain/chains/graph_qa/cypher.py [hunk 1]
```diff
@@ -176,6 +189,10 @@ def _call(
         # Extract Cypher code if it is wrapped in backticks
         generated_cypher = extract_cypher(generated_cypher)
 
+        # Correct Cypher query if enabled
+        if self.cypher_query_corrector:
+            generated_cypher = self.cypher_query_corrector(generated_cypher)
```
Comment (jexp) [inline libs/langchain/langchain/chains/graph_qa/cypher.py:194]: Should we raise a warning or error if the generated cypher is not correct? 
Should it actually become null, or rather get a warning attached with the issue pointed out?
Perhaps something for later.

Comment (tomasonjo) [reply libs/langchain/langchain/chains/graph_qa/cypher.py:194]: at the moment it becomes null, and so the context becomes automatically null.
But we only catch if the relationship direction doesn't fit the schema, we don't catch anything else.

I don't really know if raising an issue is the way to go, we can decide along the way

**Comment by tomasonjo** [comment]
@jexp probably need a final LGTM :)

**Review by jexp — APPROVED** [review]
LGTM

**Comment by baskaryan** [comment]
looks awesome, thanks @sakusaku-rich @tomasonjo @jexp!

**Pull request merged**